source/agora/network/Client.d

```diff
@@ -223,17 +223,21 @@ public class NetworkClient
 
     /***************************************************************************
 
+        Params:
+            challange = a challange for Validator to prove identity
+
         Returns:
-            the node's public key.
+            The public key of this node and signature for the challange
 
         Throws:
             `Exception` if the request failed.
 
     ***************************************************************************/
 
-    public PublicKey getPublicKey () @trusted
+    public Identity getPublicKey (ulong challange) @trusted
     {
-        return this.attemptRequest!(API.getPublicKey, Throw.Yes)(this.api);
+        return this.attemptRequest!(API.getPublicKey, Throw.Yes)(this.api,
+            challange);
     }
```

Comment:
We might want to think about the challenge for a bit, because we're essentially offering an endpoint to sign a random 8 bytes... That could be dangerous. Luckily there's a proven way to securely exchange keys... Diffie Hellman.

Reply:
Very good point, thanks. What about this one?

Reply:
It's still susceptible to replay attacks, isn't it ?

Reply:
Yes:( Let's just append the random numbers instead of adding them. Pushing in a second.